programs/fileio.c

```diff
@@ -1454,6 +1454,16 @@ FIO_compressFilename_srcFile(FIO_prefs_t* const prefs,
     ress.srcFile = FIO_openSrcFile(srcFileName);
     if (ress.srcFile == NULL) return 1;   /* srcFile could not be opened */
 
+    /* Check if "srcFile" is compressed. Only done if --exclude-compressed flag is used
+    * YES => ZSTD will not compress the file.
+    * NO => ZSTD will resume with compress operation.
+    */
+    if (g_excludeCompressedFiles && UTIL_isCompressedFile(srcFileName)) {  /* precompressed file (--exclude-compressed). DO NOT COMPRESS */
```

Comment:
minor : optimization :
could this rejection test be done _before_ `FIO_openSrcFile()` ?
This would avoid a needless `fopen()` / `fclose()` cycle.

Reply:
Addressed in latest commit.